### ISSUE TITLE
visibility issues

### ISSUE THREAD

**Issue by nudowding** [abrisco/rules_pyo3#16]
I created a simple package that builds a pyo3_extension. I can depend on that target in the same package in a py_library or py_binary target, but cannot access the extension in a different package.

I get an error like 

```
py_binary rule //bin/foo:foo: target '//lib/bar_py:bar_py' is not visible from target '//bin/foo:foo'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

I've tried these settings individually in  `//lib/bar_py`  and none of them work. I would expect them all to work.

global setting for the package 

`package(default_visibility = ["//visibility:public"])`

visibility setting for extension e.g. visibility = ...
-  `["//:__subpackages__"]`
-  `["//bin/foo:__pkg__"]`
- `["//visibility:public"]`

What does work is setting the package visibility **and** setting the visibility in the extension rule. I haven't tried exhaustive combinations, but values that are the same or compatible seem to work like `["//visibility:public"]` for the package and the rule or  `["//visibility:public"]` for the package and `["//:__subpackages__"]` for the rule.

**Comment by abrisco** [comment]
@nudowding thanks for the bug report! I believe https://github.com/abrisco/rules_pyo3/pull/19 should fix this issue.

**Comment by nudowding** [comment]
Thanks for the quick fix @abrisco,

I tried 0.0.6 and it worked as expected.

From: Andre Brisco ***@***.***>
Date: Thursday, September 12, 2024 at 5:35 PM
To: abrisco/rules_pyo3 ***@***.***>
Cc: George Dowding ***@***.***>, Mention ***@***.***>
Subject: Re: [abrisco/rules_pyo3] visibility issues (Issue #16)
Closed #16 [github. com] as completed via #19 [github. com]. — Reply to this email directly, view it on GitHub [github. com], or unsubscribe [github. com]. You are receiving this because you were mentioned. Message ID: <abrisco/rules_pyo3/issue/16/issue_event/14240539600@ github. com>
ZjQcmQRYFpfptBannerStart
CAUTION: External Email

ZjQcmQRYFpfptBannerEnd

Closed #16 [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_abrisco_rules-5Fpyo3_issues_16&d=DwMFaQ&c=s883GpUCOChKOHiocYtGcg&r=EgBvvLiXOJgTmwdpANcKV-W5g-hvGfD5yBGPeTgMvCc&m=thlzWzeI-mCpEsoqCr9ghPYFhSm75n4YbQkpvPZ19Qal8TnKey1711Q8iSeeoWMc&s=Xsf0E6xKvNQP8MPIEn5CEDBIxmnFU2YpraENS4RSGpU&e=> as completed via #19 [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_abrisco_rules-5Fpyo3_pull_19&d=DwMFaQ&c=s883GpUCOChKOHiocYtGcg&r=EgBvvLiXOJgTmwdpANcKV-W5g-hvGfD5yBGPeTgMvCc&m=thlzWzeI-mCpEsoqCr9ghPYFhSm75n4YbQkpvPZ19Qal8TnKey1711Q8iSeeoWMc&s=HwEXbDtkjhFCEzUvTN0MZPhE9PX5RXzlFTdqaV8Ocqc&e=>.
